src/umpire/tpl/judy/judyL2Array.h

```diff
@@ -217,7 +217,7 @@ class judyL2Array {
         bool removeEntry( JudyKey key ) {
             if( 0 != ( _lastSlot = ( vector ** ) judy_slot( _judyarray, ( const unsigned char * ) &key, _depth * JUDY_key_size ) ) ) {
                 // _lastSlot->~vector(); //for use with placement new
-                // delete _lastSlot;
+                delete *_lastSlot;
```

Comment:
@rrsettgast this leak in particular is what caused the issue in geos.

Reply:
@corbett5, thanks for finding this!  What was the 'issue in geos'?

Reply:
if ` _lastSlot` is of type `vector **`, then wouldn't `delete _lastSlot` only delete the `vector**`, leaving the `vector*` hanging out? I assume that was the "leak" we were seeing. However, I can't tell where is the `new` that corresponds with this `delete`. @corbett5 I would dig around a bit to confirm what is the "correct" thing to do....on the other hand, you don't have any more leaks....soooooo....yeah.

Reply:
@mcfadden8 A developer created a `chai::ManagedArray` in a tight loop and then ran out of memory on his laptop. When he changed the `chai::ManagedArray` to a vector the memory usage went back to normal. A simple reproducer is to use Umpire to allocate and deallocate memory in an infinite loop then look at the memory usage of the process over time.

@rrsettgast  You'll notice that the call to `delete _lastSlot` was previously commented out. However the line above that is equivalent to the line I added when using placement new.
The corresponding `new`s are here
https://github.com/LLNL/Umpire/blob/725284aad29641c6754f7ca81f4c94d4e5e4cb2a/src/umpire/tpl/judy/judyL2Array.h#L105
and here
https://github.com/LLNL/Umpire/blob/725284aad29641c6754f7ca81f4c94d4e5e4cb2a/src/umpire/tpl/judy/judyL2Array.h#L130

You can see that `delete *_lastSlot` is called in `clear`
https://github.com/LLNL/Umpire/blob/725284aad29641c6754f7ca81f4c94d4e5e4cb2a/src/umpire/tpl/judy/judyL2Array.h#L68

Which with my change makes `removeEntry` basically the inner loop of `clear` which certainly makes sense to me.

Reply:
looks correct.